cider/exec/nextgen/parsers/RelAlgEUParser.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "exec/nextgen/parsers/Parser.h"
+
+#include "exec/nextgen/operators/FilterNode.h"
+#include "exec/nextgen/operators/ProjectNode.h"
+#include "exec/nextgen/operators/SourceNode.h"
+#include "util/Logger.h"
+
+namespace cider::exec::nextgen::parsers {
+using namespace cider::exec::nextgen::operators;
+
+static bool isParseable(const RelAlgExecutionUnit& eu) {
+  if (!eu.join_quals.empty()) {
+    LOG(ERROR) << "JOIN is not supported in RelAlgExecutionUnitParser.";
+    return false;
+  }
+  if (eu.groupby_exprs.empty()) {
+    LOG(ERROR) << "Aggregation is not supported in RelAlgExecutionUnitParser.";
```

Comment:
I think you need add comments here. Actually the logic is tricky even we know it.

Reply:
we should remove this trick at some point in the future.